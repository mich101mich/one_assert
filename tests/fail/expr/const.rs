fn main() {
    one_assert::assert!(const { 3 + 1 });
    one_assert::assert!(
        const {
            let x = 1;
            x == 1;
        }
    );
}
