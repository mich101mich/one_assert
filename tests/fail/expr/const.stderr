error[E0308]: mismatched types
 --> tests/fail/expr/const.rs:2:33
  |
2 |     one_assert::assert!(const { 3 + 1 });
  |                                 ^^^^^ expected `bool`, found integer

error[E0308]: mismatched types
 --> tests/fail/expr/const.rs:4:15
  |
4 |           const {
  |  _______________^
5 | |             let x = 1;
6 | |             x == 1;
  | |                   - help: remove this semicolon to return this value
7 | |         }
  | |_________^ expected `bool`, found `()`
